cli/spec.py

```diff
@@ -227,6 +267,124 @@ def __init__(self, value):
         self.value = res_dict
 
 
+class ArgumentsPreProcessor(object):
+    """
+    The helper class which is responsible to to transform input cli arguments
+    prior passing them to the clg module for parsing.
```

Comment:
Please add an example here to further explain what it does.
Give an input example and the transformation